libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -645,6 +661,10 @@ void AP_GPS::update_instance(uint8_t instance)
         if (state[instance].status >= GPS_OK_FIX_2D) {
             timing[instance].last_fix_time_ms = tnow;
         }
+
+        if (should_df_log() && !AP::ahrs().have_ekf_logging()) {
+            DataFlash_Class::instance()->Log_Write_GPS(instance);
```

Comment:
This needs to be done in the case when we delete the driver as well, as we need a log message for when we give up on a GPS. The previous version caught that via the resulting timestamp shift.